activestorage/app/models/active_storage/blob.rb

```diff
@@ -27,6 +27,8 @@ class ActiveStorage::Blob < ActiveRecord::Base
 
   has_many :attachments
 
+  scope :unattached, -> { left_joins(:attachments).where(ActiveStorage::Attachment.table_name => { blob_id: nil }) }
```

Comment:
Don't we need an `index` in `blob_id` now? The index we have has `:record_type`, `:record_id` and `:name` before the blob_id so the left-most rule will not apply anymore.

Reply:
There is an index already https://github.com/rails/rails/blob/4c615a53e0409ddddf13d48cbc328a22d0026ed2/activestorage/db/migrate/20170806125915_create_active_storage_tables.rb#L18.
`.references` adds index by default.

Reply:
oh cool! I always forget that